### PR TITLE
[JEP-229] Fix Jenkinsfile to provide credentials; enable CD for log-cli

### PR DESCRIPTION
**Pull request by daniel-beck**
Follow up on https://github.com/jenkins-infra/repository-permissions-updater/pull/1747 with some unfinished WIP I had forgotten about 😢 

Anyway, this seems to work now, tested by switching the job on trusted.ci over to this origin branch (and back again).

@jglick clearly is fine with `log-cli` being CD enabled, so that's why I set that up here already.

### CODE REVIEW

File: permissions/plugin-log-cli.yml [hunk 1]
```diff
@@ -1,6 +1,8 @@
 ---
 name: "log-cli"
 github: "jenkinsci/log-cli-plugin"
+cd:
+  enabled: true
```
Comment (jglick) [inline permissions/plugin-log-cli.yml:5]: :+1:

**Review by jglick — APPROVED** [review]

**Comment by daniel-beck** [comment]
FTR this is deliberately still undocumented, I don't think we're quite ready to declare this JEP active.

**Comment by jglick** [comment]
> I don't think we're quite ready to declare this JEP active.

Yes, can still be in **Draft** status until some more pieces are implemented and tested. @oleg-nenashev I would still really like to get https://github.com/jenkinsci/jep/pull/332 merged though, so that the published edition more closely reflects the various changes that are being made.

**Review by timja — APPROVED** [review]

**Comment by jglick** [comment]
I see `MAVEN_USERNAME` + `MAVEN_TOKEN` in https://github.com/jenkinsci/log-cli-plugin/settings/secrets/actions, great…

**Comment by jglick** [comment]
https://github.com/jglick/jenkins-maven-cd-action/issues/2 accordingly.

**Comment by timja** [comment]
This ready for use in other plugins @jglick?

**Comment by jglick** [comment]
@timja yes if you are ready to live with very experimental stuff; it is up and running in `log-cli-plugin` anyway. I suspect many maintainers will want https://github.com/jglick/jenkins-maven-cd-action/issues/1, and https://github.com/jglick/verify-ci-status-action/issues/1 is likely to become a priority if changes proposed on jenkins-dev go ahead, and I plan to request `HOSTING` to get these tools into @jenkinsci or @jenkins-infra. For the near term I would recommend use on relatively minor plugins to minimize the impact of any grave problems that might arise.

I actually think that non-plugin components (`bom`, `jenkins-test-harness`) would be good candidates too—there is nothing about this system that is limited to `*-plugin` that I know of. Just need to be incrementalified, deployed to Artifactory, and managed via RPU, typically via `org.jenkins-ci:jenkins`. (`archetypes` and other components deployed to OSSRH, typically via `org.kohsuke:pom`, are not eligible. `plugin-pom` is also probably not incrementalifiable for technical reasons.)

**Comment by timja** [comment]
Experimental is fine with me :)

**Pull request merged**